cpp/cycles_module/algorithm/cycles.hpp

```diff
@@ -11,7 +11,7 @@ namespace cycles_util {
 struct NodeState {
   std::vector<bool> visited;
   std::vector<std::uint64_t> discovery, depth;
-  std::vector<std::uint64_t> parent;
+  std::vector<std::int64_t> parent;
```

Comment:
@antonio2368 What do you think of this. This should be a container that keeps the track of parent nodes within the zero-indexed nodes in the graph. The problem is the initialization of that container, the right thing to do is setting it to -1, but it can be accomplished only by using `int64_t` which is incompatible to the `uint64_t` in which nodes are stored.

Reply:
`std::optional<uint64_t>` :smile: